kimchi/src/circuits/polynomials/permutation.rs

```diff
@@ -56,11 +56,16 @@ use rand::{CryptoRng, RngCore};
 
 /// Number of constraints produced by the argument.
 pub const CONSTRAINTS: u32 = 3;
-pub const ZK_ROWS: u64 = 3;
+/// Number of randomized rows (1 per evaluation).
+pub const ZK_ROWS: u64 = 2;
+/// Number of rows for the final accumulator value.
+pub const ACC_ROWS: u64 = 1;
+/// Total number of evaluation rows.
+pub const EVAL_ROWS: u64 = ZK_ROWS + ACC_ROWS;
 /// Evaluates the polynomial
 /// (x - w^{n - 4}) (x - w^{n - 3}) * (x - w^{n - 2}) * (x - w^{n - 1})
 pub fn eval_vanishes_on_last_4_rows<F: FftField>(domain: D<F>, x: F) -> F {
-    let w4 = domain.group_gen.pow(&[domain.size - (ZK_ROWS + 1)]);
+    let w4 = domain.group_gen.pow(&[domain.size - (EVAL_ROWS + 1)]);
```

Comment:
maybe this value here should be called `PERM_FINAL_CONSTRAINTS`, as it represents the final constraints as explained here https://o1-labs.github.io/proof-systems/plonk/zkpm.html

so I think it'd be make sense to define something like


`const PERM_FINAL_CONSTRAINTS: usize = EVALS /* zk rows */ + 1 /* end of accumulator */;`